scalafix-core/shared/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -300,53 +289,39 @@ object EscapeHatch {
     def apply(
         tree: Tree,
         associatedComments: AssociatedComments): AnchoredEscapes = {
-      val enableBuilder = TreeMap.newBuilder[EscapeOffset, List[EscapeFilter]]
-      val disableBuilder = TreeMap.newBuilder[EscapeOffset, List[EscapeFilter]]
-      val unusedAnchoredEnable = List.newBuilder[Position]
+      val enable = TreeMap.newBuilder[EscapeOffset, List[EscapeFilter]]
+      val disable = TreeMap.newBuilder[EscapeOffset, List[EscapeFilter]]
       val visitedFilterExpression = mutable.Set.empty[Position]
-      var currentlyDisabledRules = Map.empty[String, Position]
-
-      def enable(
-          rules: String,
-          start: EscapeOffset,
-          anchor: Token.Comment): Unit =
-        enableBuilder += (start -> makeFilters(start, None, anchor, rules))
-
-      def disable(
-          rules: String,
-          start: EscapeOffset,
-          end: Option[EscapeOffset],
-          anchor: Token.Comment): Unit =
-        disableBuilder += (start -> makeFilters(start, end, anchor, rules))
+      val onOffTracker = new OnOffTracker
 
       def makeFilters(
+          rulesPos: List[(String, Position)],
           start: EscapeOffset,
           end: Option[EscapeOffset],
-          anchor: Token.Comment,
-          rules: String): List[EscapeFilter] = {
-        val splitRules0 = splitRules(rules)
-
-        if (splitRules0.isEmpty) { // wildcard
+          anchor: Comment): List[EscapeFilter] = {
+        if (rulesPos.isEmpty) { // wildcard
           List(
             EscapeFilter(FilterMatcher.matchEverything, anchor.pos, start, end))
         } else {
-          rulesWithPosition(splitRules0, anchor).map {
+          rulesPos.map {
             case (rule, pos) =>
               EscapeFilter(FilterMatcher(rule), pos, start, end)
           }
         }
       }
 
-      def splitRules(rules: String): List[String] =
-        rules.trim.split("\\s*,\\s*").toList
+      def splitRules(rules: String): List[String] = {
+        val trimmed = rules.trim
+        if (trimmed.isEmpty) Nil else trimmed.split("\\s*,\\s*").toList
+      }
```

Comment:
this method had a bug that was accidentally making the unused reporting logic work for some cases.